packages/erxes-ui/src/components/richTextEditor/hooks/useExtensions.ts

```diff
@@ -46,75 +46,15 @@ import {
   getMentionSuggestions,
 } from '../utils/getMentionSuggestions';
 import { generateJSON } from '@tiptap/html';
-
 export type UseExtensionsOptions = {
   /** Placeholder hint to show in the text input area before a user types a message. */
   placeholder?: string;
-
   showMentions?: boolean;
-
   mentionSuggestion?: MentionSuggestionParams;
-
   /** Character count limit. */
   limit?: number;
 };
 
-// Don't treat the end cursor as "inclusive" of the Link mark, so that users can
-// actually "exit" a link if it's the last element in the editor (see
-// https://tiptap.dev/api/schema#inclusive and
-// https://github.com/ueberdosis/tiptap/issues/2572#issuecomment-1055827817).
-// This also makes the `isActive` behavior somewhat more consistent with
-// `extendMarkRange` (as described here
-// https://github.com/ueberdosis/tiptap/issues/2535), since a link won't be
-// treated as active if the cursor is at the end of the link. One caveat of this
-// approach: it seems that after creating or editing a link with the link menu
-// (as opposed to having a link created via autolink), the next typed character
-// will be part of the link unexpectedly, and subsequent characters will not be.
-// This may have to do with how we're using `insertContent` and `setLink` in
-// the LinkBubbleMenu, but I can't figure out an alternative approach that
-// avoids the issue. This is arguably better than being "stuck" in the link
-// without being able to leave it, but it is still not quite right. See the
-// related open issues here:
-// https://github.com/ueberdosis/tiptap/issues/2571,
-// https://github.com/ueberdosis/tiptap/issues/2572, and
-// https://github.com/ueberdosis/tiptap/issues/514
-const CustomLinkExtension = Link.extend({
-  inclusive: false,
-  addAttributes() {
-    return {
-      href: {
-        default: null,
-      },
-      target: {
-        default: this.options.HTMLAttributes.target,
-      },
-      rel: {
-        default: this.options.HTMLAttributes.rel,
-      },
-      style: {
-        parseHTML: (element) => element.getAttribute('style'),
-        renderHTML: (attributes) => {
-          if (!attributes.style) {
-            return {};
-          }
-          return { style: attributes.style };
-        },
-      },
-    };
-  },
-});
-// Make subscript and superscript mutually exclusive
-// https://github.com/ueberdosis/tiptap/pull/1436#issuecomment-1031937768
-
-/// @later config these
-const CustomSubscript = Subscript.extend({
-  excludes: 'superscript',
-});
-
-const CustomSuperscript = Superscript.extend({
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_refinement):** Consider simplifying the extension customization for readability and maintenance.

The extensive comments and custom configurations for `Link`, `Subscript`, and `Superscript` extensions could be simplified or better organized for easier readability and maintenance. Consider encapsulating these customizations in separate functions or files.